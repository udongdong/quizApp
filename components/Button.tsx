import React from 'react';
import {Pressable, StyleSheet, Text, View, ViewStyle} from 'react-native';
import {color} from '../styles';

type ButtonType = 'primary' | 'secondary';

type ButtonProps = {
  style?: ViewStyle;
  type?: ButtonType;
  title: string;
  onPress: () => void;
};

export function Button(props: ButtonProps): React.JSX.Element {
  const {style, type = 'primary', title, onPress} = props;

  const btnStyle = type === 'primary' ? styles.primary : styles.secondary;
  const btnTextStyle =
    type === 'primary' ? styles.primaryText : styles.secondaryText;

  return (
    <View style={{...styles.box, ...style}}>
      <Pressable
        style={{...styles.container, ...btnStyle, ...style}}
        onPress={onPress}>
        <Text style={{...btnTextStyle}}>{title}</Text>
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  box: {
    width: '100%',
    height: 52,
    alignSelf: 'center',
  },
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    width: '100%',
  },
  primary: {
    backgroundColor: color.primary,
    borderWidth: 0,
  },
  primaryText: {
    color: color.white,
  },
  secondary: {
    backgroundColor: color.gray,
    borderWidth: 1,
    borderColor: color.primary,
  },
  secondaryText: {
    color: color.primary,
  },
});
