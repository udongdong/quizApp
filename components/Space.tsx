import React from 'react';
import {View} from 'react-native';

type SpaceProps = {
  h?: number;
  w?: number;
};

export function Space(props: SpaceProps): React.JSX.Element {
  const {h = 0, w = 0} = props;

  return <View style={{marginTop: h, marginLeft: w}} />;
}
