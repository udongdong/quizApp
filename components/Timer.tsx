import React, {useEffect, useState} from 'react';
import {StyleSheet, Text} from 'react-native';

export function Timer(): React.JSX.Element {
  const [time, setTime] = useState<number>(0);

  useEffect(() => {
    const timer = setTimeout(() => setTime(t => t + 1), 1000);
    return () => clearTimeout(timer);
  });

  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  return (
    <Text style={styles.container}>
      {`${minutes}`.padStart(2, '0')}:{`${seconds}`.padStart(2, '0')}
    </Text>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    padding: 10,
    width: '19%',
  },
});
