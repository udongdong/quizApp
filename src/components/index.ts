export * from './Button';
export * from './ProgressBar';
export * from './RadioButton';
export * from './Space';
export * from './quiz';
