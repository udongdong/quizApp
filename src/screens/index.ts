export * from './MainScreen';
export * from './MainTab';
export * from './quiz';
export * from './review';
export * from './screenTypes';
