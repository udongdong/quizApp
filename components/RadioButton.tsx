import React, {useState} from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';
import {Button} from './Button';

type RadioButtonProps = {
  style?: ViewStyle;
  radios: string[];
  init?: number;
  onSelected?: (i: number) => void;
};

export function RadioButton(props: RadioButtonProps): React.JSX.Element {
  const {style, radios, init = 0, onSelected} = props;
  const [selected, setSelected] = useState<number>(init);

  return (
    <View style={{...styles.container, ...style}}>
      {radios.map((r, i) => {
        const type = selected === i ? 'primary' : 'secondary';
        return (
          <Button
            style={{...styles.button, ...{flex: 1}}}
            title={r}
            onPress={() => {
              setSelected(i);

              if (onSelected) {
                onSelected(i);
              }
            }}
            type={type}
          />
        );
      })}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  button: {},
});
