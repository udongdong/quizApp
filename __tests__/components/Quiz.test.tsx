import 'react-native';
import React from 'react';
import {Quiz, QuizProps} from '../../src/components';

import {render, RenderAPI} from '@testing-library/react-native';

function getComponent(props: QuizProps) {
  return <Quiz {...props} />;
}

describe('Quiz Component test (퀴즈 모드)', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    const props: QuizProps = {
      question: 'In Yu-Gi-Oh, how does a player perform an Xyz Summon?',
      answers: [
        `Banish A Number of Monsters From Your Hand And Deck`,
        `Add the Monsters' Levels Together to Match the Xyz Monster`,
        `Overlay at least 2 Monsters of the Same Level`,
        `Activate a Spell and Send Monsters to the Graveyard`,
      ],
      difficulty: 'medium',
      correct: 'Overlay at least 2 Monsters of the Same Level',
      isDone: false,
    };

    rendered = render(getComponent(props));
  });

  test('render Quiz Component (퀴즈 모드)', () => {
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  test('question 이 잘 세팅된다.', () => {
    expect(
      rendered.getByText(
        'Q) In Yu-Gi-Oh, how does a player perform an Xyz Summon?',
      ),
    ).toBeDefined();
  });

  test('난이도에 맞는 뱃지가 잘 세팅된다.', () => {
    expect(rendered.getByText('medium')).toBeDefined();
  });
});

describe('Quiz Component test (리뷰 모드)', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    const props: QuizProps = {
      question: 'In Yu-Gi-Oh, how does a player perform an Xyz Summon?',
      answers: [
        'Banish A Number of Monsters From Your Hand And Deck',
        `Add the Monsters' Levels Together to Match the Xyz Monster`,
        'Overlay at least 2 Monsters of the Same Level',
        'Activate a Spell and Send Monsters to the Graveyard',
      ],
      difficulty: 'medium',
      correct: 'Overlay at least 2 Monsters of the Same Level',
      selected: 'Banish A Number of Monsters From Your Hand And Deck',
      isDone: true,
    };

    rendered = render(getComponent(props));
  });

  test('render Quiz Component (리뷰 모드)', () => {
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  test('정답 표시가 잘 세팅된다.', () => {
    expect(
      rendered.getByText(
        'Overlay at least 2 Monsters of the Same Level (정답)',
      ),
    ).toBeDefined();
  });

  test('난이도에 맞는 뱃지가 잘 세팅된다.', () => {
    expect(rendered.getByText('medium')).toBeDefined();
  });
});
