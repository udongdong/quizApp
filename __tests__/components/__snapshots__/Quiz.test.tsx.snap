// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Component test (리뷰 모드) render Quiz Component (리뷰 모드) 1`] = `
<View
  style={
    {
      "margin": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F58F16",
        "borderColor": "orange",
        "borderRadius": 25,
        "borderWidth": 3,
        "height": 50,
        "justifyContent": "center",
        "margin": 10,
        "width": 50,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 10,
          "fontWeight": "bold",
        }
      }
    >
      medium
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 20,
      }
    }
  >
    Q) In Yu-Gi-Oh, how does a player perform an Xyz Summon?
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          {
            "color": "red",
          },
        ]
      }
    >
      Banish A Number of Monsters From Your Hand And Deck
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          {},
        ]
      }
    >
      Add the Monsters' Levels Together to Match the Xyz Monster
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          {
            "color": "#5F37FF",
          },
        ]
      }
    >
      Overlay at least 2 Monsters of the Same Level (정답)
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          {},
        ]
      }
    >
      Activate a Spell and Send Monsters to the Graveyard
    </Text>
  </View>
</View>
`;

exports[`Quiz Component test (퀴즈 모드) render Quiz Component (퀴즈 모드) 1`] = `
<View
  style={
    {
      "margin": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F58F16",
        "borderColor": "orange",
        "borderRadius": 25,
        "borderWidth": 3,
        "height": 50,
        "justifyContent": "center",
        "margin": 10,
        "width": 50,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 10,
          "fontWeight": "bold",
        }
      }
    >
      medium
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 20,
      }
    }
  >
    Q) In Yu-Gi-Oh, how does a player perform an Xyz Summon?
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          false,
        ]
      }
    >
      Banish A Number of Monsters From Your Hand And Deck
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          false,
        ]
      }
    >
      Add the Monsters' Levels Together to Match the Xyz Monster
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          false,
        ]
      }
    >
      Overlay at least 2 Monsters of the Same Level
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Image
      source={{}}
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 10,
          },
          false,
        ]
      }
    >
      Activate a Spell and Send Monsters to the Graveyard
    </Text>
  </View>
</View>
`;
